internal/pkg/opentelemetry/controller.go

```diff
@@ -110,3 +112,153 @@ func (c *Controller) Shutdown(ctx context.Context) error {
 	}
 	return nil
 }
+
+func appendAttrs(dest []attribute.KeyValue, m pcommon.Map) []attribute.KeyValue {
+	m.Range(func(k string, v pcommon.Value) bool {
+		dest = append(dest, attr(k, v))
+		return true
+	})
+	return dest
+}
+
+func attr(k string, v pcommon.Value) attribute.KeyValue {
+	return attribute.KeyValue{
+		Key:   attribute.Key(k),
+		Value: val(v),
+	}
+}
+
+func val(val pcommon.Value) (out attribute.Value) {
```

Comment:
All these conversion functions will be removed once we move to a collector receiver, right?

Reply:
Yes! That is the plan.